05/DB.test.js

```diff
@@ -1 +1,131 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe('insert', () => {
+    it('', async () => {
+        expect.assertions(1);
+
+        const db = new DB();
+        await db.insert({ id: 1 , name: 'dog'})
+        await db.insert({ id: 2, name: 'cat'})
+
+        const rows = await db.getRows();
+
+        expect(rows.length).toBe(2)
+    });
+
+    it('error', async () => {
+        expect.assertions(1);
+
+        const db = new DB();
+
+        try {
+            await db.insert({ id: 'a', name: 'dog'})
+            await db.insert({ id: 2, name: 'cat'})
+        } catch(e) {
+            expect(e).toBe('ID can be only number!')
+        }
+    });
+
+    it('error', async () => {
+        expect.assertions(1);
+
+        const db = new DB();
+
+        try {
+            await db.insert({ id: 2, name: 'dog'})
+            await db.insert({ id: 2, name: 'cat'})
+        } catch(e) {
+            expect(e).toBe('ID can\'t be duplicated!')
+        }
+    });
+
+    it('add ID if it is not written', async () => {
+        const db = new DB();
+        const insert =  {name: 'fish'}
+        const newData = await db.insert(insert)
+
+        expect(newData).toEqual(insert)
+    });
+});
+
+describe('select', () => {
+    it('item with specific ID is in row', async () => {
+        expect.assertions(1);
+
+        const db = new DB();
+        await db.insert({ id: 1, name: 'dog'})
+        const checkForId = await db.select(1)
+        expect(checkForId).toEqual({ id: 1, name: 'dog'})
```

Comment:
Dla pewności utworzył bym zmienną `const data = { id: 1, name: 'dog'}` i nią bym operował.
Wtedy nie ma możliwości popełnić błąd. Jak piszmy z "palca" to coś zawsze może nie pójść :P